### PR TITLE
chore: WS-2491 Greenhouse token and drupal cleanup

### PR DESCRIPTION
**Pull request by marekwrobel**
Simple cleanup - removes the greenhouse token and PROSPECTUS_PROXY_PASS (this was used when we were on Drupal).

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A SRE team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a SRE ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/spaces/SRE/pages/28967861/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/prospectus/defaults/main.yml [hunk 1]
```diff
@@ -19,15 +19,13 @@ PROSPECTUS_OAUTH_SECRET: !!null
...
 
-PROSPECTUS_GREENHOUSE_API_TOKEN: 'fake-key'
 PROSPECTUS_CONTENTFUL_SPACE_ID: 'fake-key'
 PROSPECTUS_CONTENTFUL_ACCESS_TOKEN: 'fake-key'
 PROSPECTUS_SEGMENT_WRITE_KEY: 'fake-key'
 PROSPECTUS_ALGOLIA_BROWSE_KEY: 'fake-key'
 
 # nginx vars
 PROSPECTUS_DATA_DIR: '/edx/var/prospectus'
-PROSPECTUS_PROXY_PASS: 'http://localhost:8080'
```
Comment (albemarle) [inline playbooks/roles/prospectus/defaults/main.yml:30]: Why is this being taken out?

**Review by albemarle — CHANGES_REQUESTED** [review]
See comments.

File: playbooks/roles/prospectus/defaults/main.yml (hunk 1 above)
Comment (marekwrobel) [reply playbooks/roles/prospectus/defaults/main.yml:30]: As per ticket description, PROSPECTUS_PROXY_PASS  was used when we were on Drupal and should also be cleaned up.

**Review by jesseday — APPROVED** [review]

File: playbooks/roles/prospectus/defaults/main.yml (hunk 1 above)
Comment (albemarle) [reply playbooks/roles/prospectus/defaults/main.yml:30]: 🤦‍♀️  Right.  :-)

**Review by albemarle — APPROVED** [review]

**Pull request merged**